packages/test-utils/src/file-system.ts

```diff
@@ -99,3 +100,48 @@ const ROOT_PATH = resolvePath(fileURLToPath(import.meta.url), '../../../..');
 export function getPathFromRoot(path: string): string {
   return resolvePath(ROOT_PATH, path);
 }
+
+/**
+ * Remove a directory and all its contents.
+ *
+ * @param path - The path to the directory to remove.
+ * @throws If the directory cannot be removed.
+ */
+export async function removeDirectory(path: string) {
+  try {
+    await rm(path, { recursive: true });
+  } catch (error) {
+    if (
+      typeof error === 'object' &&
+      error !== null &&
+      'code' in error &&
+      error.code !== 'ENOENT'
+    ) {
+      // eslint-disable-next-line @typescript-eslint/no-throw-literal
+      throw error;
+    }
```

Comment:
It looks like this only throws for well-formed errors that are not `ENOENT`, but wouldn't we also want to throw for malformed errors, e.g. `!('code' in error)`?